tests/pc/test_retry_count.py

```diff
@@ -0,0 +1,266 @@
+        """
+        Test that peers and DUT use new LACPDU version when peers use a non-standard retry count
+        """
+        check_lacpdu_packet_version(duthost)
+
+    def test_kill_teamd_lag_up(self, duthost, nbrhosts, higher_retry_count_on_peers, config_reload_on_cleanup):
+        """
+        Test that the lag remains up for 150 seconds after killing teamd on the peer
+        """
+        for nbr in list(nbrhosts.keys()):
+            nbrhosts[nbr]['host'].shell("sudo pkill -x teamd")
+
+        # Give ourselves 30 seconds to check before the LAG goes down. This should also handle the
+        # worst case scenario where the last LACPDU was sent 29 seconds prior to teamd getting
+        # killed.
+        time.sleep(120)
+
+        cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"]
+        port_channels = cfg_facts["PORTCHANNEL"].keys()
+        for port_channel in port_channels:
+            port_channel_status = duthost.get_port_channel_status(port_channel)
+            for _, status in list(port_channel_status["ports"].items()):
+                pytest_assert(status["runner"]["selected"], "status of lag member error")
+                pytest_assert(status["runner"]["partner_retry_count"] == 5, "partner retry count is incorrect")
+
+
+class TestDutRetryCount:
+    def test_retry_count(self, duthost, nbrhosts, higher_retry_count_on_dut):
+        """
+        Test that peers see new retry count when DUT updates retry count.
+        """
+        cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")["ansible_facts"]
+        port_channels = cfg_facts["PORTCHANNEL"].keys()
+        for port_channel in port_channels:
+            port_channel_status = duthost.get_port_channel_status(port_channel)
+            pytest_assert(port_channel_status["runner"]["retry_count"] == 5, "retry count on DUT is incorrect")
+
+        for nbr in list(nbrhosts.keys()):
+            port_channel_status = nbrhosts[nbr]['host'].get_port_channel_status("PortChannel1")
+            for _, status in list(port_channel_status["ports"].items()):
+                pytest_assert(status["runner"]["selected"], "status of lag member error")
+                pytest_assert(status["runner"]["partner_retry_count"] == 5, "partner retry count is incorrect")
+
+    def test_retry_count_packet_version(self, duthost, nbrhosts, higher_retry_count_on_dut, scapy_lacp_layer):
+        """
+        Test that peers and DUT use new LACPDU version when DUT uses a non-standard retry count
+        """
+        check_lacpdu_packet_version(duthost)
+
+    def test_kill_teamd_peer_lag_up(self, duthost, nbrhosts, higher_retry_count_on_peers, config_reload_on_cleanup):
+        """
+        Test that the lag remains up for 150 seconds after killing teamd on the DUT
+        """
+        duthost.shell("sudo pkill -x teamd")
```

Comment:
If you kill teamd with SIGUSR1 then it will ensure that the last LACPDU is sent before it dies.

`docker exec teamd pkill -USR1 teamd`

Reply:
Done.